### PR TITLE
list: added list_item_del_init

### PR DESCRIPTION
**Pull request by dabekjakub**
Added function that will delete and reinitialize list.

Reinitialized lists show up as empty unlike deleted with
list_item_del.

Signed-off-by: Jakub Dabek <jakub.dabek@linux.intel.com>

### CODE REVIEW

**Comment by lgirdwood** [comment]
@dabekjakub can you send PR for stable-1.2 too

File: src/include/sof/list.h [hunk 1]
```diff
@@ -72,13 +72,24 @@ static inline void list_item_append(struct list_item *item,
...
 	item->prev->next = item->next;
 }
 
+/* delete item from the list list iteam will be reinitialised
+ * and list_is_empty will return true
+ */
+static inline void list_item_del_init(struct list_item *item)
+{
+	list_item_del(item);
+	list_init(item);
```
Comment (xiulipan) [inline src/include/sof/list.h]: Here you init only the item not the list.
What do you mean by
> and list_is_empty will return true

Comment (jajanusz) [reply src/include/sof/list.h]: Item and list are interchangable.

> and list_is_empty will return true

When you just del item it still has pointers pointing to items in list it's not part of. When you init item it's just like newly created item that is not part of any list and will return true for list_is_empty - it doesn't check for null it next but if next == 'this item'

**Comment by xiulipan** [comment]
@lgirdwood @dabekjakub 
Why don't just add `list_init(item);`
to all `list_item_del` to make sure all del is safe?

**Comment by lgirdwood** [comment]
@xiulipan pls send a PR

**Comment by xiulipan** [comment]
@lgirdwood 
Reuse #300 and make the update.

**Pull request merged**